packages/metro-config/index.js

```diff
@@ -45,6 +45,9 @@ function getDefaultConfig(
       unstable_conditionNames: ['require', 'import', 'react-native'],
     },
     serializer: {
+      getModulesRunBeforeMainModule: () => [
```

Comment:
Note: This exists in CLI already (dependent on the `--platforms` arg), but yes it makes sense to promote this default to the base config.

https://github.com/react-native-community/cli/blob/main/packages/cli-plugin-metro/src/tools/loadMetroConfig.ts#L52C9-L54C11

Can we add a small comment above this, to point maintainers to both sources of truth? :)

```suggestion
      // Note: This option is overridden in cli-plugin-metro (getOverrideConfig)
      getModulesRunBeforeMainModule: () => [
```

Related: @motiz88 might have a plan to redesign the `getModulesRunBeforeMainModule` and `getPolyfills` options soon, to harden against these kind of issues.

Reply:
thanks for the comment to make the code clean. and feel excited for the api redesign 🔥